### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by SajjadAhmad14**
- Add filter category to Redux store.
- Use useState, useDispatch, and useSelecter hooks.

### CODE REVIEW

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi @SajjadAhmad14  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
- [ ] Kindly check again and confirm that you have submitted the correct milestone. Not a single requirement has been implemented for this milestone. If you are sure that you have submitted the correct milestone, then I would suggest that you carefully read through the project requirements again. Below is a screenshot of the milestone requirements.
Screenshot from 2021-08-02 01-58-17


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

File: src/containers/Categories.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const categories = () => {
+  const CATEGORIES = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+  return CATEGORIES;
+};
```
Comment (bellom) [inline src/containers/Categories.js]: Opps 🤦 , let's stick to best practice.

In milestone 2, you were introduced to the `component` and `container` folders.

Kindly improve the project structure such that you have components for stateless components and a container/page folder for other components that use the state.

React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory (presentational components) you have components that are not connected to the Redux store, and on your containers directory, you have components that are connected to the Redux store. Read More... https://www.pluralsight.com/guides/how-to-structure-redux-components-and-containers

In your case, it should be `src/component/Categories`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @SajjadAhmad14  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check my comments under this review;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

File: src/containers/Categories.js (hunk 1 above)
Comment (SajjadAhmad14) [reply src/containers/Categories.js]: Hi,
Thank you so much for making the presentational and stateful components clear!

**Review by mahmoud717 — APPROVED** [review]
### STATUS: APPROVED ✔️ ✔️ 

**hi @SajjadAhmad14**

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**

**Pull request merged**